examples/distributed/graphsage/node_classification.py

```diff
@@ -50,40 +68,43 @@ def forward(self, blocks, x):
 
     def inference(self, g, x, batch_size, device):
         """
-        Inference with the GraphSAGE model on full neighbors (i.e. without
-        neighbor sampling).
-
-        g : the entire graph.
-        x : the input of entire node set.
-
-        Distributed layer-wise inference.
+        Distributed layer-wise inference with the GraphSAGE model on full
+        neighbors.
+
+        Parameters
+        ----------
+        g : DistGraph
+            Input Graph for inference.
+        x : DistTensor
+            Node feature data of input graph.
+
+        Returns
+        -------
+        DistTensor
+            Inference results.
         """
-        # During inference with sampling, multi-layer blocks are very
-        # inefficient because lots of computations in the first few layers
-        # are repeated. Therefore, we compute the representation of all nodes
-        # layer by layer.  The nodes on each layer are of course splitted in
-        # batches.
-        # TODO: can we standardize this?
+        # Split nodes to each trainer.
         nodes = dgl.distributed.node_split(
             np.arange(g.num_nodes()),
             g.get_partition_book(),
             force_even=True,
         )
-        y = dgl.distributed.DistTensor(
-            (g.num_nodes(), self.n_hidden),
-            th.float32,
-            "h",
-            persistent=True,
-        )
+
         for i, layer in enumerate(self.layers):
+            # Create DistTensor to save forward results.
             if i == len(self.layers) - 1:
-                y = dgl.distributed.DistTensor(
-                    (g.num_nodes(), self.n_classes),
-                    th.float32,
-                    "h_last",
-                    persistent=True,
-                )
-            print(f"|V|={g.num_nodes()}, eval batch size: {batch_size}")
+                out_dim = self.n_classes
+                name = "h_last"
+            else:
+                out_dim = self.n_hidden
+                name = "h"
+            y = dgl.distributed.DistTensor(
+                (g.num_nodes(), out_dim),
+                th.float32,
+                name,
+                persistent=True,
+            )
+            print(f"|V|={g.num_nodes()}, inference batch size: {batch_size}")
 
             sampler = dgl.dataloading.NeighborSampler([-1])
```

Comment:
I believe adding a brief description here would make it easier for those less familiar to understand its meaning. Especially the interpretation of the parameter `[-1]`.

Reply:
added